coriolis/tasks/replica_tasks.py

```diff
@@ -554,23 +555,36 @@ def run(self, ctxt, instance, origin, destination, task_info,
             raise exception.CoriolisException(
                 "Replica source provider plugin for '%s' does not support"
                 " updating Replicas" % origin["type"])
-
-        origin_connection_info = base.get_connection_info(ctxt, origin)
-        volumes_info = task_info.get("volumes_info", {})
-
-        LOG.info("Checking source provider environment params")
-        # NOTE: the `source_environment` in the `origin` is the one set
-        # in the dedicated DB column of the Replica and thus stores
-        # the previous value of it:
-        old_source_env = origin.get('source_environment', {})
-        volumes_info = (
-            source_provider.check_update_source_environment_params(
-                ctxt, origin_connection_info, instance, volumes_info,
-                old_source_env, new_source_env))
-        schemas.validate_value(
-            volumes_info, schemas.CORIOLIS_VOLUMES_INFO_SCHEMA)
-
-        task_info['volumes_info'] = volumes_info
+        # All replica update errors for the source environment should be
+        # silenced if 'force' param is applied.
+        try:
+            origin_connection_info = base.get_connection_info(ctxt, origin)
+            volumes_info = task_info.get("volumes_info", {})
+
+            LOG.info("Checking source provider environment params")
+            # NOTE: the `source_environment` in the `origin` is the one set
+            # in the dedicated DB column of the Replica and thus stores
+            # the previous value of it:
+            old_source_env = origin.get('source_environment', {})
+            volumes_info = (
+                source_provider.check_update_source_environment_params(
+                    ctxt, origin_connection_info, instance, volumes_info,
+                    old_source_env, new_source_env))
+            schemas.validate_value(
+                volumes_info, schemas.CORIOLIS_VOLUMES_INFO_SCHEMA)
+
+            task_info['volumes_info'] = volumes_info
+        except Exception as ex:
+            if task_info['force']:
+                event_manager.progress_update(
+                    'Forced Replica Update is executed, ignoring: %s' % (
+                        str(ex)))
+                LOG.trace(utils.get_exception_details())
+            else:
+                raise exception.CoriolisException(
```

Comment:
1) Please just `LOG.warn` this message and re-raise the initial exception to keep its trace.
2) In the future, never format `utils.get_exception_details()` into exception messages themselves as it will lead to the whole ugly-ass trace being exposed through the API.